lesson17/script.js

```diff
@@ -0,0 +1,147 @@
+
+const init = async () => {
+    showLoading();
+    try{
+        const sliderData = await fetchSliderData(3000);
+        if (!sliderData) {
+            return;
+        }
+        renderSliderItems(sliderData);
+        renderPagenation(sliderData);
+        
+        renderButtons();
+        isActiveFirstSliderItem();
+        SwitchSliderItem();
+    }catch(error){
+        console.error(error);
+        createErrorMessage(error);
+    }finally{
+        removeLoading();
+    }
+}
+
+const renderSliderItems = (sliderData) => {
+    const fragment = document.createDocumentFragment();
+    for(const [index,image] of sliderData.entries()){
+        const sliderItem = createElement({tagName:"div",className:"slider__item"});
+        const slideImage = createImage({className:"slide-image", src:image.img, alt:image.alt});
+        sliderItem.style.zIndex = `${sliderData.length -+ index}`;
+        
+        fragment.appendChild(sliderItem).appendChild(slideImage);
+    }
+    slider.appendChild(fragment);
+}
+
+const renderButtons = () => {
+    const prevBtn = createButton({id:"js-prevBtn",className:"js-btn prev-btn slider-btn"});
+    const nextBtn = createButton({id:"js-nextBtn",className:"js-btn next-btn slider-btn"});
+    prevBtn.disabled = "true";
+
+    slider.appendChild(prevBtn).insertAdjacentElement("afterend",nextBtn);
+}
+
+const renderPagenation = (sliderData) => {
+    const pagenation    = createElement({tagName:"p",className:"pagenation",text:`/${sliderData.length}`});
+    const currentNumber = createElement({tagName:"span",id:"js-currentNumber",className:"current-number",text:"1"});
+
+    slider.appendChild(pagenation).insertAdjacentElement("afterbegin",currentNumber);
+    return pagenation;
+}
+
+const isActiveFirstSliderItem = () => document.getElementsByClassName("slider__item")[0].classList.add("is-active");
+
+const SwitchSliderItem = () => {
+    const sliderButtons = document.getElementsByClassName("js-btn");
+    const sliderItems   = document.getElementsByClassName("slider__item");
+
+    let index = 0;
+    for (const button of sliderButtons ){
+        button.addEventListener("click",function(){      
+            sliderItems[index].classList.remove("is-active");
+
+            button.id === "js-nextBtn" ? index++ : index--;
+            sliderItems[index].classList.add("is-active");
+
+            changeCurrentNumber(index);
+            disabledButton(sliderItems,index);
+        });
+    }
+}
+
+const changeCurrentNumber = (index) => document.getElementById("js-currentNumber").textContent = `${index +1}`;
+
+const disabledButton = (sliderItems,index) => {
+    const nextBtn    = document.getElementById("js-nextBtn");
+    const prevBtn    = document.getElementById("js-prevBtn");
+    const firstIndex = 0;
+    const lastIndex  = sliderItems.length -1;
+
+    prevBtn.disabled = index === firstIndex;
+    nextBtn.disabled = index === lastIndex;
+
+    return lastIndex;
+}
```

Comment:
nits 
If a button is disabled, how about changing the color of the button？With the current UI, I felt that the button could be pressed even if disabled.

Reply:
I agree with your opinion.
I will try to set pointerevents:none;.

Reply:
I fixed.
https://github.com/nakagawamai/js-lesson/pull/19/commits/8964b04ba73e3498e3e5facda4ca9610dfbbd46d